src/declarations/stencil-public-compiler.ts

```diff
@@ -367,19 +376,26 @@ export interface ConfigExtras {
    */
   slotChildNodesFix?: boolean;
 
-  /**
-   * Enables the tagNameTransform option of `defineCustomElements()`, so the component tagName
-   * can be customized at runtime. Defaults to `false`.
-   */
-  tagNameTransform?: boolean;
-
   // TODO(STENCIL-914): remove `experimentalSlotFixes` when it's the default behavior
   /**
    * Enables all slot-related fixes such as {@link slotChildNodesFix}, and
    * {@link scopedSlotTextContentFix}.
    */
-  experimentalSlotFixes?: boolean;
-}
+  experimentalSlotFixes?: false;
+}
+
+export type ConfigExtras = ConfigExtrasBase &
+  (
+    | ConfigExtrasSlotFixes
+    | {
+        // TODO(STENCIL-914): remove `experimentalSlotFixes` when it's the default behavior
+        /**
+         * Enables all slot-related fixes such as {@link slotChildNodesFix}, and
+         * {@link scopedSlotTextContentFix}.
+         */
+        experimentalSlotFixes?: true;
+      }
+  );
```

Comment:
So, with this approach, even a config like the following would be invalid via the TS language server, but wouldn't throw a compiler warning:

```ts
extras: {
  experimentalSlotFixes: true,
  slotChildNodesFix: false // <-- doesn't like that this is specified
}
```

This may be confusing. We could, alternatively, have the types setup where if `experimentalSlotFixes: true` is set, then all other flags _could_ be specified, but would need to be `false`:

```ts
extras: {
  experimentalSlotFixes: true,
  slotChildNodesFix: false // <-- now this would be valid
}
```

Likewise, the logic for `experimentalSlotFixes: false`, would work as it does with the current implementation.

Making this change could revert the changes in [this](https://github.com/ionic-team/stencil/pull/4767/files#diff-bd7059cb44e3fa48f56fceff1b07ccba76c002943745485cf28d32b81449e024) file and reduce/remove the `any` casts in the [validation tests](https://github.com/ionic-team/stencil/pull/4767/files#diff-164622140dac3d1aa088d2b1c0f6b8de0630cf3b715c25b301c54208f5d3b5bf)

Lemme know what you guys think.

Reply:
> This may be confusing. We could, alternatively, have the types setup where if experimentalSlotFixes: true is set, then all other flags could be specified, but would need to be false

I like this approach better, but I'm also happy with how it's working ATM - so I suppose I only lean towards this approach (requiring if the other flags are set, they have to be false) for removing type assertions